camviewer_ui_impl.py

```diff
@@ -2810,3 +2880,21 @@ def atomic_writer(path: str) -> typing.Iterator[typing.TextIO]:
     # If the tempfile still exists, we should clean it up.
     if os.path.exists(fd.name):
         os.remove(fd.name)
+
+
+def decode_char_waveform(waveform: npt.NDArray[np.int8]) -> str:
+    """
+    Convert an epics char waveform to a string.
+
+    In pyca, these can be loaded into numpy arrays via passing
+    numpy=True as a kwarg.
+
+    The waveform is an array of signed 8-bit integers whose
+    unsigned representations correspond to the ascii character codes.
+    The string is null-terminated.
+    """
+    # Implementation lifted from PyDM's "parse_value_for_display"
+    zeros = np.where(waveform == 0)[0]
+    if zeros.size > 0:
+        waveform = waveform[: zeros[0]]
```

Comment:
I'll be honest, I didn't explore improving this snippit at all.

I think since the EPICS string is null-terminated, any data after the first 0 is invalid by definition.

Reply:
Yeah, I think in general we expect something like

123 120 142 0 0 0 0 0 0

And we should be able to treat data after the first 0 as "garbage" by definition

But I'm not sure if we ever get these sorts of strings in practice